### PR TITLE
[TEVA-2135] Only put organisation_id in session

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-2135

### CODE REVIEW

File: spec/system/hiring_staff_can_add_stats_to_a_vacancy_spec.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@
   let(:school) { create(:school) }
 
   before(:each) do
```
Comment (cpjmcquillan) [inline spec/system/hiring_staff_can_add_stats_to_a_vacancy_spec.rb:9]: We can lose the redundant `(:each)`

File: spec/system/hiring_staff_have_to_accept_terms_spec.rb [hunk 2]
```diff
@@ -2,13 +2,12 @@
 
 RSpec.describe "Hiring staff accepts terms and conditions" do
   let(:school) { create(:school) }
-  let(:oid) { "a-valid-oid" }
-  let(:current_publisher) { Publisher.find_by(oid: oid) }
+  let(:current_publisher) { Publisher.first }
```
Comment (cpjmcquillan) [inline spec/system/hiring_staff_have_to_accept_terms_spec.rb:5]: Where's the first publisher coming from?

Comment (david-mears-dfe) [reply spec/system/hiring_staff_have_to_accept_terms_spec.rb:5]: E.g. line 10.

File: app/services/publishers/vacancy_filter.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 class Publishers::VacancyFilter
   attr_reader :managed_school_ids, :managed_organisations
 
-  def initialize(publisher, school_group)
+  def initialize(publisher, organisation)
+    school_group = (organisation if organisation.is_a?(SchoolGroup))
```
Comment (cpjmcquillan) [inline app/services/publishers/vacancy_filter.rb:5]: I'm assuming this is is one of the things you're going to refactor next - but if not let's refactor it!

**Review by cpjmcquillan — APPROVED** [review]
💥

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3070.london.cloudapps.digital>

File: app/services/publishers/vacancy_filter.rb (hunk 3 above)
Comment (cesidio) [reply app/services/publishers/vacancy_filter.rb:5]: This is going to change a lot in the next piece of work, will refactor it later

**Pull request merged**